azure-quantum/azure/quantum/_workspace_connection_params.py

```diff
@@ -53,13 +66,19 @@ def __init__(
         client_id: Optional[str] = None,
         client_secret: Optional[str] = None,
         api_version: Optional[str] = None,
+        api_key: Optional[str] = None,
+        connection_string: Optional[str] = None,
         on_new_client_request: Optional[Callable] = None,
     ):
         self._location = None
         self._environment = None
         self._base_url = None
         self._arm_base_url = None
 
+        # connection_string is set first as it
+        # should be overridden by other parameters
+        self.apply_connection_string(connection_string)
```

Comment:
Just out of curiosity, what happens if we receive a valid connection string, and in the environment variables or somewhere else we set partial of the parameters. Will the final connection_parameters be a mix?

Reply:
Just in the case of environment variables. Individual parameters from the connection string will be used if that parameter is not already set via other environment variables.

If the user creates the workspace from the connection string, the environment variables are not used at all.